src/middlewared/middlewared/plugins/smb.py

```diff
@@ -1313,7 +1301,16 @@ async def apply_presets(self, data):
         from preset if user-defined aux parameters already exist. In this
         case user-defined takes precedence.
         """
-        params = (SMBSharePreset[data["purpose"]].value)["params"].copy()
+        if data["afp"]:
+            params = {
+                'acl': True,
+                'aapl_name_mangling': True,
+                'streams': True,
+                'durablehandle': False,
+                'auxsmbconf': '',
+            }
+        else:
+            params = (SMBSharePreset[data["purpose"]].value)["params"].copy()
```

Comment:
Back out this change now that we're handling in registry.py

Reply:
@anodos325 do you mean that we need to apply
```
                'acl': True,
                'aapl_name_mangling': True,
                'streams': True,
                'durablehandle': False,
``` in registry.py too ignoring databse options? But we disable `acl` for some of the migrated shares, that will break it.

Reply:
No, sorry, my eyes glazed over for a minute. But we do not need to overlay these items via a preset. Setting them to during initial migration is enough.

The only item that must enabled is the encoding. This we set via `fruit:encoding = native` in registry as I suggested elsewhere.

Reply:
> Setting them to during initial migration is enough.

How will these options be forced for newly created AFP shares then?

Reply:
There won't be any newly created AFP shares. If user creates an SMB share with `afp` checked, then the only minor deviation from above will be status of `durablehandle`, which based on some issues users have seen, I may change default of.  `durablehandle = false` is not a requirement for compatibility with legacy AFP shares. It only affects whether our server advertises support for durable handles for that share (ability to survive network disruptions).

Setting `afp` and having `aapl_extensions` is sufficient to force the correct character encoding settings for the share.